src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -360,6 +358,53 @@ private static void AddProjectFileDependenciesForPackageReference(PackageSpec pr
             }
         }
 
+        private void AddProjectCentralTransitiveDependencyGroupsForPackageReference(PackageSpec project, LockFile lockFile, IEnumerable<RestoreTargetGraph> targetGraphs)
+        {
+            // TargetFrameworkInformations should have only distinct FrameworkNames
+            var centralPackageVersionsPerFramework = project.TargetFrameworks.ToDictionary(tfmi => tfmi.FrameworkName, tfmi => tfmi.CentralPackageVersions);
+
+            foreach (var targetGraph in targetGraphs.OrderBy(graph => graph.Framework.ToString(), StringComparer.Ordinal))
+            {
+                if (_includeFlagGraphs.TryGetValue(targetGraph, out Dictionary<string, LibraryIncludeFlags> dependenciesIncludeFlags))
+                {
+                    // The transitive dependencies enforced by the central package version management file are written to the assets to be used by the pack task.
+                    IEnumerable<LibraryDependency> centralEnforcedTransitiveDependencies = targetGraph
+                        .Flattened
+                        .Where(graphItem => graphItem.IsCentralTransitive
+                            && centralPackageVersionsPerFramework.ContainsKey(targetGraph.Framework)
```

Comment:
This probably shouldn't be failing, likely a redundant check, the rest of the code should be accounting for this, probably the validator or something like that.

Reply:
It is correct, refactored the code.